rct229/rulesets/ashrae9012019/section1/section1rule1.py

```diff
@@ -0,0 +1,142 @@
+                ),
+                rmrs_used_optional=produce_ruleset_model_instance(
+                    USER=True,
+                    BASELINE_0=True,
+                    BASELINE_90=True,
+                    BASELINE_180=True,
+                    BASELINE_270=True,
+                    PROPOSED=True,
+                ),
+                manual_check_required_msg=MANUAL_CHECK_REQUIRED_MSG,
+            )
+
+        def get_calc_vals(self, context, data=None):
+            rmd_u = context.USER
+            rmd_b0 = context.BASELINE_0
+            rmd_b90 = context.BASELINE_90
+            rmd_b180 = context.BASELINE_180
+            rmd_b270 = context.BASELINE_270
+            rmd_p = context.PROPOSED
+
+            output_bpf_set = []
+
+            for rmd in (rmd_u, rmd_b0, rmd_b90, rmd_b180, rmd_b270, rmd_p):
+                if rmd is not None:
+                    output_bpf_set.append(
+                        find_one(
+                            "$.output.total_area_weighted_building_performance_factor",
+                            rmd,
+                        )
+                    )
+
+            output_bpf_set = list(filter(lambda x: x is not None, output_bpf_set))
+            assert_(
+                len(output_bpf_set) >= 1,
+                "At least one `output_bpf_set` value must exist.",
+            )
+            bpf_building_area_type_dict = get_BPF_building_area_types_and_zones(rmd_b0)
+            is_undetermined = "UNDETERMINED" in bpf_building_area_type_dict.keys()
+            bpf_bat_sum_prod = 0
+            total_area = ZERO.AREA
+            climate_zone = data["climate_zone"]
+            for bpf_bat in bpf_building_area_type_dict.keys():
+                if is_undetermined:
+                    continue
+                expected_bpf = table_4_2_1_1_lookup(bpf_bat, climate_zone)[
+                    "building_performance_factor"
+                ]
+                total_area += bpf_building_area_type_dict[bpf_bat]["area"]
+                bpf_bat_sum_prod += (
+                    expected_bpf * bpf_building_area_type_dict[bpf_bat]["area"]
+                )
+            return {
+                "output_bpf_set": list(set(output_bpf_set)),
+                "bpf_bat_sum_prod": bpf_bat_sum_prod,
+                "total_area": total_area,
+                "is_undetermined": is_undetermined,
+            }
+
+        def manual_check_required(self, context, calc_vals=None, data=None):
+            is_undetermined = calc_vals["is_undetermined"]
+            return is_undetermined
+
+        def rule_check(self, context, calc_vals=None, data=None):
+            output_bpf_set = calc_vals["output_bpf_set"]
+            bpf_bat_sum_prod = calc_vals["bpf_bat_sum_prod"]
+            total_area = calc_vals["total_area"]
+
+            return (
+                len(output_bpf_set) == 1
+                and output_bpf_set[0] != 0
+                and bpf_bat_sum_prod / total_area == output_bpf_set[0]
```

Comment:
We should use the ```std_equal``` function instead of ```==```.

```Python
std_equal(output_bpf_set[0], bpf_bat_sum_prod / total_area)
```

Reply:
Good point! Addressed